### PR TITLE
compliance/add_sanction_list_providers

### PR DESCRIPTION
**Pull request by heshan-deriv**
Add a package to communicate with the collectordb.
https://app.clickup.com/t/20696747/COMP-1460

### CODE REVIEW

File: lib/Data/Sanctions/DB.pm [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                    {Slice => {}},
+                    $p_provider_name, $p_provider_url, $p_publish_date, $p_hash, $p_number_of_entries
+                );
+            });
+    } catch {
+        warn "Failed to insert or update sanction list provider: $_";
+        $success = 0;
+    };
+
+    return $success;
```
Comment (saad-deriv) [inline lib/Data/Sanctions/DB.pm:95]: Just a suggestion. `return 0;` in the catch and then return 1; in the end.

File: lib/Data/Validate/Sanctions/Fetcher.pm [hunk 2]
```diff
@@ -528,4 +542,65 @@ sub _entries_from_remote_src {
...
+        $data,
+        {
+            canonical => 1,
+            utf8      => 1
+        });
+
+    # Generate and return the SHA-256 hash
+    return sha256_hex($json_string);
+}
+
```
Comment (saad-deriv) [inline lib/Data/Validate/Sanctions/Fetcher.pm:605]: Nicely done. Should have done something similar for my stuff as we are doing similar computing in almost all the xml parsing.

**Review by saad-deriv — APPROVED** [review]

**Review by bassam-deriv — APPROVED** [review]

File: lib/Data/Validate/Sanctions/Fetcher.pm [hunk 3]
```diff
@@ -528,4 +542,65 @@ sub _entries_from_remote_src {
...
+  my $hash = $fetcher->create_hash($data);
+
+=cut
+
+sub _create_hash {
+    my ($data) = @_;
+
+    # Convert the data to a JSON string
+    my $json_string = to_json(
```
Comment (ragheb-deriv) [inline lib/Data/Validate/Sanctions/Fetcher.pm:595]: why are we converting it to JSON, is there a specific reason

Comment (heshan-deriv) [reply lib/Data/Validate/Sanctions/Fetcher.pm:595]: `$data `is a hash which is structured data. I could convert the hash to a string by concatenating the key-values but converting it to json make it more standardized.

File: lib/Data/Sanctions/DB.pm (hunk 1 above)
Comment (heshan-deriv) [reply lib/Data/Sanctions/DB.pm]: This is fixed.

**Review by ragheb-deriv — APPROVED** [review]

**Review by bassam-deriv — APPROVED** [review]

**Review by chylli-deriv — APPROVED** [review]

**Pull request merged**